qiskit/pulse/instructions/play.py

```diff
@@ -75,6 +75,22 @@ def duration(self) -> Union[int, ParameterExpression]:
         """Duration of this instruction."""
         return self.pulse.duration
 
+    def _initialize_parameter_table(self,
+                                    operands: Tuple[Any]):
```

Comment:
Why does this function have different implementations for different instructions?

Reply:
Because parameters are stored in different place and the parameter table usually defines a mapping of parameter object to where it's stored, i.e. index of operand tuple.

For example, shift phase
```
operands = (phase, channel)
```
When we parameterize phase and channel, this will generate a table `{phase: [0], channel: [1]}`. This is the standard logic to create the table.

However, in Call instruction
```
operands = (subroutine, )
```
and the `subroutine` cannot be `ParameterExpression` object. If we use the same logic, the table always becomes `{amp: [0], phase: [0], ...}`. Since value is not stored in the parameter table, we need to assign values immediately with object copy. This is huge overhead. Instead, I define a parameter table as something like `{amp: 0.1, phase: 0.5}`, i.e. a mapping of object to assigned value. This is more convenient because we can later assign parameters in `inline_subroutine` function with this table. Object copy is not performed at every parameter assignment.

Reply:
Thanks.